sdk/Pulumi/Provider/Provider.cs

```diff
@@ -916,14 +966,156 @@ public override async Task<Empty> Delete(Pulumirpc.DeleteRequest request, Server
             }
         }
 
-        public override Task<Pulumirpc.ConstructResponse> Construct(Pulumirpc.ConstructRequest request, ServerCallContext context)
+        public override async Task<Pulumirpc.ConstructResponse> Construct(Pulumirpc.ConstructRequest request, ServerCallContext context)
+        {
+            try
+            {
+                var aliases = request.Aliases.Select(urn => (Input<Alias>)new Alias() { Urn = urn }).ToList();
+
+                InputList<Resource> dependsOn = request.Dependencies
+                    .Select(urn => new DependencyResource(urn))
+                    .ToImmutableArray<Resource>();
+                var providers = request.Providers.Values
+                    .Select(reference => new DependencyProviderResource(reference))
+                    .ToList<ProviderResource>();
+
+                var opts = new ComponentResourceOptions()
+                {
+                    Aliases = aliases,
+                    DependsOn = dependsOn,
+                    Protect = request.Protect,
+                    Providers = providers,
+                    Parent = request.Parent != null ? new DependencyResource(request.Parent) : null,
+                };
+
+                var domRequest = new ConstructRequest(request.Name, request.Type, Marshal(request.Inputs), opts);
```

Comment:
This needs to get the resource monitor endpoint from the request and setup everything for using that.